### DISCUSSION TITLE
Sankey diagrams

### DISCUSSION THREAD

**Discussion by alixander** [terrastruct/d2#604] [Ideas]
<img width="596" alt="Screen Shot 2023-01-04 at 10 28 34 AM" src="https://user-images.githubusercontent.com/3120367/210624425-8c84d8b6-0e3e-4c51-aa03-a9c14674b496.png">

<img width="691" alt="Screen Shot 2023-01-04 at 10 28 30 AM" src="https://user-images.githubusercontent.com/3120367/210624441-0f8313ed-ba9e-4dc2-a497-5be49ec03e48.png">

https://www.reddit.com/r/golang/comments/zq8f9h/comment/j2xfzy7/?utm_source=share&utm_medium=web2x&context=3

**Comment by alixander** [comment]
D2's syntax should be fine for it. e.g.

```
shape: sankey

wages
wages -> housing: 420
wages -> taxes: 450
```

will see if anyone else wants for this before putting it up as todo.

**Reply by jtorvald** [reply]
Would the current syntax also provide a way to change unit type and colors?

**Reply by alixander** [reply]
yeah. honestly it's like 90% of the way there, just needs a custom layout engine like we did for sequence diagrams:

https://play.d2lang.com/?script=SipNSU8tUdC1U8jILy3OzEu3UqjmUlAAe9pKwdDAgEtBAWwNRBzEQTgCpABJ0EohvygxLz2VS0GhlquWiwthdkliRWoxqsmmhEw2RTG5KDUFaiwgAAD__w%3D%3D&layout=elk

**Reply by 8Almd5bx** [reply]
Alixander, I am a medical student who needs help the sankey diagram for one of our project.
Can I get professional help from you if you provide this service?
I will appreciate if you could email me if you are interested in please. (karakas186@gmail.com)
Thank you!
